ios/Sources/Repository/Authenticator.swift

```diff
@@ -0,0 +1,52 @@
+//
+//  Authenticator.swift
+//  
+//
+//  Created by watanavex on 2021/06/12.
+//
+
+import Foundation
+import DroidKaigiMPP
+import FirebaseAuth
+
+class AuthenticatorImpl: Authenticator {
+
+    func currentUser(completionHandler: @escaping (DroidKaigiMPP.User?, Error?) -> Void) {
+        guard let firebaseUser = Auth.auth().currentUser else {
+            completionHandler(nil, nil)
+            return
+        }
+        firebaseUser.getIDTokenResult { (result, error) in
+            if let error = error {
+                completionHandler(nil, error)
+                return
+            }
+
+            completionHandler(DroidKaigiMPP.User(idToken: result?.token), nil)
+        }
+    }
+
+    func signInAnonymously(completionHandler: @escaping (DroidKaigiMPP.User?, Error?) -> Void) {
+        Auth.auth().signInAnonymously { (result, error) in
+            if let error = error {
+                completionHandler(nil, error)
+                return
+            }
+
+            guard let result = result else {
+                completionHandler(DroidKaigiMPP.User(idToken: nil), nil)
```

Comment:
Is this successful state? It seems like failure state.

Reply:
https://github.com/firebase/firebase-ios-sdk/blob/e7a078c4c35735bdbd338585801a63e480ebfa87/FirebaseAuth/Sources/Backend/FIRAuthBackend.m#L853-L857
It looks like an impossible case, so it may be better to add an assertion failure?

Reply:
Yes, I agree.

@watanavex 
Please add `assertionFailure` to this block 🙏🏼 
And even if this is impossible case, this should be handled correctly.
What does passing `DroidKaigiMPP.User(idToken: nil)` to `completionHandler` mean?
I think if `result` is `nil`, this shouldn't pass `DroidKaigi.User` instance to `completionHandler` .

Reply:
I changed the definition in Kotlin.
User with nil token is useless... 🤔 
I've also improved Auth API.

37c87f8